File: app/src/main/java/com/mtp/laboproject/data/repository/AlertsRepository.java
package com.mtp.laboproject.data.repository;

import com.mtp.laboproject.data.model.alert.AlertsResponse;
import com.mtp.laboproject.data.remoteApi.ApiInterface;

import io.reactivex.Single;

public class AlertsRepository {

    private ApiInterface api;

    public AlertsRepository(ApiInterface api) {
        this.api = api;
    }
    public Single<AlertsResponse> getAlertsRx(){
         return api.getAlerts();
    }


}

File: app/src/main/java/com/mtp/laboproject/data/repository/AlertsRepository.kt
package com.mtp.laboproject.data.repository


import com.mtp.laboproject.data.model.AlertsDetailsResponse
import com.mtp.laboproject.data.remoteApi.ApiInterface


class AlertsRepository(private val api: ApiInterface) : BaseRepository() {

    //get latest news using safe api call
    suspend fun getAlerts(): MutableList<AlertsDetailsResponse>? {
        return safeApiCall(
            //await the result of deferred type
            call = { api.getAlerts().await() },
            error = "Error fetching news"
            //convert to mutable list
        )?.toMutableList()
    }
}